framework/auth/__init__.py

```diff
@@ -153,6 +154,7 @@ def get_or_create_institutional_user(fullname, sso_email, sso_identity, primary_
     # Note: Institution users are created as confirmed with a strong and random password. Users don't need the
     # password since they sign in via SSO. They can reset their password to enable email/password login.
     user = OSFUser.create_confirmed(sso_email, str(uuid.uuid4()), fullname)
+    user.add_system_tag(institution_source_tag(primary_institution._id))
```

Comment:
In addition to this one, we need to add system tags for the secondary institution if one exists.

I think this is the place you want to add it: https://github.com/CenterForOpenScience/osf.io/blob/3920a29ff5c92a45229e618931661cc5d0c8a08d/api/institutions/authentication.py#L384

Reply:
I added it down here: https://github.com/CenterForOpenScience/osf.io/pull/10696/files#diff-0f4ded4c84c49e7558006cdd622c8cbbc49883203ea7348bcaf8d186b5c7a804R393 where the secondary institution is already defined.